packages/aws-cdk/lib/notices.ts

```diff
@@ -0,0 +1,247 @@
+    const expiration = cachedData.expiration ?? 0;
+
+    if (Date.now() > expiration) {
+      const freshData = {
+        expiration: Date.now() + TIME_TO_LIVE,
+        notices: await this.dataSource.fetch(),
+      };
+      await this.save(freshData);
+      return freshData.notices;
+    } else {
+      return notices;
+    }
+  }
+
+  private async load(): Promise<CachedNotices> {
+    try {
+      return await fs.readJSON(this.fileName) as CachedNotices;
+    } catch (e) {
+      debug(`Failed to load notices from cache: ${e}`);
+      return {
+        expiration: 0,
+        notices: [],
+      };
+    }
+  }
+
+  private async save(cached: CachedNotices): Promise<void> {
+    try {
+      await fs.writeJSON(this.fileName, cached);
+    } catch (e) {
+      debug(`Failed to store notices in the cache: ${e}`);
+    }
+  }
+}
+
+export interface NoticeFilterProps {
+  permanentlySuppressed: boolean,
+  temporarilySuppressed: boolean,
+  cliVersion: string,
+  frameworkVersion: string | undefined,
+  acknowledgedIssueNumbers: Set<number>,
+}
+
+export class NoticeFilter {
+  private readonly acknowledgedIssueNumbers: Set<number>;
+
+  constructor(private readonly props: NoticeFilterProps) {
+    this.acknowledgedIssueNumbers = props.acknowledgedIssueNumbers;
+  }
+
+  /**
+   * Returns true if we should show this notice.
+   */
+  apply(notice: Notice): boolean {
+    if (this.props.permanentlySuppressed
+      || this.props.temporarilySuppressed
+      || this.acknowledgedIssueNumbers.has(notice.issueNumber)) {
+      return false;
+    }
+    return this.applyVersion(notice, 'cli', this.props.cliVersion) ||
+      this.applyVersion(notice, 'framework', this.props.frameworkVersion);
+  }
+
+  /**
+   * Returns true if we should show the notice.
+   */
+  private applyVersion(notice: Notice, name: string, compareToVersion: string | undefined) {
+    if (compareToVersion === undefined) { return true; }
+
+    const affectedComponent = notice.components.find(component => component.name === name);
+    const affectedVersion = affectedComponent?.version;
+    return affectedVersion != null && semver.satisfies(compareToVersion, affectedVersion);
+  }
+}
+
+export interface NoticeFormatter {
+  apply(notice: Notice): string,
+}
+
+// TODO rename this
+export class FooNoticeFormatter implements NoticeFormatter {
+  apply(notice: Notice): string {
+    const componentsValue = notice.components.map(c => `${c.name}: ${c.version}`).join(', ');
+    return [
+      `${notice.issueNumber}\t${notice.title}`,
+      `\tOverview: ${notice.overview}`,
+      `\tAffected versions: ${componentsValue}`,
+      `\tMore information at: https://github.com/aws/aws-cdk/issues/${notice.issueNumber}`,
+    ].join('\n\n');
+  }
+}
+
+function frameworkVersion(outdir: string): string | undefined {
+  const tree = loadTree().tree;
+  // v2
+  if (tree?.constructInfo?.fqn.startsWith('aws-cdk-lib')) {
+    return tree.constructInfo.version;
+  }
+  // v1
+  // TODO
```

Comment:
My v1 app's `tree.json` does not seem to have the framework version listed. Not sure if there is a way around this, but we can always add an additional file to the cloud assembly that stores the metadata we need.

Reply:
I tested with version 1.144.0 and it's there: 

```
"constructInfo": {
  "fqn": "@aws-cdk/core.App",
  "version": "1.144.0"
}
```